src/controllers/auth.controller.js

```diff
@@ -0,0 +1,128 @@
+import bcrypt from 'bcrypt';
+import { userService } from '../services/user.service.js';
+import { User } from '../models/user.js';
+import { jwtService } from '../services/jwt.service.js';
+import { tokenService } from '../services/token.service.js';
+import { ApiError } from '../exceptions/api.error.js';
+import { validateEmail, validatePassword } from '../utils/methods.js';
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  await userService.registerUser(name, email, hashedPassword);
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+
+  if (!activationToken) {
+    throw ApiError.badRequest('activation token expired');
```

Comment:
The error message 'activation token expired' might be misleading if the token is simply missing. Consider rephrasing it to indicate that the activation token is required.